### DISCUSSION TITLE
TypeError: string argument without an encoding for move_cartesian_path

### DISCUSSION THREAD

**Discussion by Kai-X-Org** [HumanSupportRobot/QandA#62] [Q&A]
Hi! When I try to call whole_body.move_cartesian_path, I keep on getting a TypeError. The full error message is:

In [9]: whole_body.move_cartesian_path([geometry.pose(z=1.0)], 'hand_palm_link')
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
/opt/ros/noetic/bin/ihsrb in <module>
----> 1 whole_body.move_cartesian_path([geometry.pose(z=1.0)], 'hand_palm_link')

/opt/ros/noetic/lib/python3/dist-packages/hsrb_interface/joint_group.py in move_cartesian_path(self, waypoints, ref_frame_id)
    854             origin_to_pose2 = geometry.multiply_tuples(origin_to_ref,
    855                                                        waypoints[i])
--> 856             plan = self._plan_cartesian_path(origin_to_pose1,
    857                                              origin_to_pose2,
    858                                              odom_to_robot_pose,

/opt/ros/noetic/lib/python3/dist-packages/hsrb_interface/joint_group.py in _plan_cartesian_path(self, origin_to_pose1, origin_to_pose2, odom_to_robot_pose, initial_joint_state, collision_env)
    801         # Goal constraint
    802         tsr_g = TaskSpaceRegion()
--> 803         tsr_g.end_frame_id = bytes(self.end_effector_frame)
    804         tsr_g.origin_to_tsr = geometry.tuples_to_pose(origin_to_pose2)
    805         tsr_g.tsr_to_end = geometry.tuples_to_pose(geometry.pose())

TypeError: string argument without an encoding

In [10]: whole_body.end_effector_frame
Out[10]: 'hand_palm_link'

In [11]: whole_body.move_cartesian_path([geometry.pose(z=1.0)], ref_frame_id=None)
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
/opt/ros/noetic/bin/ihsrb in <module>
----> 1 whole_body.move_cartesian_path([geometry.pose(z=1.0)], ref_frame_id=None)

/opt/ros/noetic/lib/python3/dist-packages/hsrb_interface/joint_group.py in move_cartesian_path(self, waypoints, ref_frame_id)
    854             origin_to_pose2 = geometry.multiply_tuples(origin_to_ref,
    855                                                        waypoints[i])
--> 856             plan = self._plan_cartesian_path(origin_to_pose1,
    857                                              origin_to_pose2,
    858                                              odom_to_robot_pose,

/opt/ros/noetic/lib/python3/dist-packages/hsrb_interface/joint_group.py in _plan_cartesian_path(self, origin_to_pose1, origin_to_pose2, odom_to_robot_pose, initial_joint_state, collision_env)
    801         # Goal constraint
    802         tsr_g = TaskSpaceRegion()
--> 803         tsr_g.end_frame_id = bytes(self.end_effector_frame)
    804         tsr_g.origin_to_tsr = geometry.tuples_to_pose(origin_to_pose2)
    805         tsr_g.tsr_to_end = geometry.tuples_to_pose(geometry.pose())

TypeError: string argument without an encoding

It seems that it s trying to convert `self.end_effector_frame` into bytes, but the variable is just a string. Are we supposed to do anything special about that variable in order to call this function? Thanks a lot!

**Comment by hsr-admin** [comment]
> --> 803 tsr_g.end_frame_id = bytes(self.end_effector_frame)
The problem is in the bytes conversion.

I was able to confirm that it works by modifying as follows
`tsr_g.end_frame_id = self.end_effector_frame`

There is one more location, but we will fix it in the next release.
